src/mapmvt.c

```diff
@@ -543,12 +543,34 @@ int msMVTWriteTile(mapObj *map, int sendheaders) {
     features_size = FEATURES_INCREMENT_SIZE;
 
     msInitShape(&shape);
-    while ((status = msLayerNextShape(layer, &shape)) == MS_SUCCESS) {
+    int nshapes;
+    nshapes = layer->resultcache ? layer->resultcache->numresults
```

Comment:
There is a potential performance implication in doing msLayerGetShapeCount() (when layer->resultcache == NULL). Couldn't we avoid it in that situation to just use msLayerNextShape() as before ?

Reply:
Not sure if it is good practice, but we could entirely rely on msLayerGetShape returned value.